routes/v1.js

```diff
@@ -625,35 +658,45 @@ api.declare({
     return;
   }
 
+  // Parse timestamps
+  let created = new Date(taskDef.created);
+  let deadline = new Date(taskDef.deadline);
+  let expires = new Date(taskDef.expires);
+
   // Insert entry in deadline queue (garbage entries are acceptable)
-  var deadline = new Date(taskDef.deadline);
   await this.queueService.putDeadlineMessage(taskId, deadline);
 
+  // Ensure we have a self-dependency, this is how defineTask works now
+  if (!_.includes(taskDef.dependencies, taskId)) {
+    taskDef.dependencies.push(taskId);
```

Comment:
@imbstack, so funny/scary story, this created a problem because our current validator just sets the default value from the json-schema without making a clone.

Hence, by doing `dependencies.push(...` I actually modified the default value inside the validator.